packages/block-library/src/paragraph/style.scss

```diff
@@ -37,3 +37,10 @@ p.has-background {
 p.has-text-color a {
 	color: inherit;
 }
+
+// Specific to the empty paragraph placeholder:
+// when shown on mobile and in nested contexts, the plus to add blocks shows up on the right.
+// This padding makes sure it doesn't overlap text.
+.editor-rich-text__tinymce[data-is-placeholder-visible="true"] + .editor-rich-text__tinymce.wp-block-paragraph {
```

Comment:
It should be in `editor.scss` right?

Reply:
Argh, yes it should. Many apologies, I know you're busy. https://github.com/WordPress/gutenberg/pull/12106/commits/81a81a47a9d3b3ecb38baadcd85a3de020df95ca should fix that up.